src/core/Akka.Remote/Endpoint.cs

```diff
@@ -500,9 +500,12 @@ protected void Gated(bool writerTerminated, bool earlyUngateRequested)
                 {
                     if (earlyUngateRequested)
                         Self.Tell(new Ungate());
+                    else
```

Comment:
`ReliableDeliverySupervisor` bugfix here - when we switch from a `Receiving` behavior to a `Gated` behavior (common, after an unplanned termination) we will wait for verification that the `EndpointWriter` (child of the RDS) is `Terminated` and then either immediately `Ungate` ourselves or schedule an `Ungate` to occur in the future - the default setting is 5 seconds.

In previous implementations of this code I mis-wrote this `if` statement so the `Ungate` operation was never scheduled, causing the node to be permanently gated. That would definitely cause rejoin issues. This fixes that.